BPA/Ch.04/GLM1.stan

```diff
@@ -0,0 +1,38 @@
+data {
+  int<lower=0> nobs;            // Number of observed data
+  int<lower=0> nmis;            // Number of missing data
+  int<lower=0> nyear;           // Number of years
+  int<lower=0> nsite;           // Number of sites
+  int<lower=0> obs[nobs];       // Observed counts
+  int<lower=0> obsyear[nobs];   // Years in observed data
+  int<lower=0> obssite[nobs];   // Sites in observed data
+  int<lower=0> misyear[nmis];   // Years in missing data
+  int<lower=0> missite[nmis];   // Sites in missing data
+}
+
+parameters {
+  real alpha[nsite];
+}
+
+transformed parameters {
+  real log_lambda[nyear, nsite];
+
+  for (i in 1:nyear)
+    for (j in 1:nsite)
+      log_lambda[i, j] <- alpha[j];
+}
+
+model {
+  // Priors
+  alpha ~ normal(0, 10);    // Site effects
+
+  for (i in 1:nobs)
+    obs[i] ~ poisson_log(log_lambda[obsyear[i], obssite[i]]);
```

Comment:
This is same as above. The authors seem to prefer consistency rather than efficiency.